extensions/smallrye-graphql/deployment/src/main/java/io/quarkus/smallrye/graphql/deployment/SmallryeGraphqlProcessor.java

```diff
@@ -0,0 +1,434 @@
+
+    /**
+     * The configuration
+     */
+    SmallRyeGraphQLConfig quarkusConfig;
+
+    @ConfigRoot(name = "smallrye-graphql")
+    static final class SmallRyeGraphQLConfig {
+        /**
+         * The rootPath under which queries will be served. Default to /graphql
+         */
+        @ConfigItem(defaultValue = "/graphql")
+        String rootPath;
+
+        /**
+         * The path where GraphQL UI is available.
+         * <p>
+         * The value `/` is not allowed as it blocks the application from serving anything else.
+         */
+        @ConfigItem(defaultValue = "/graphql-ui")
+        String rootPathUi;
+
+        /**
+         * Always include the UI. By default this will only be included in dev and test.
+         * Setting this to true will also include the UI in Prod
+         */
+        @ConfigItem(defaultValue = "false")
+        boolean alwaysIncludeUi;
+
+        /**
+         * If GraphQL UI should be enabled. By default, GraphQL UI is enabled.
+         */
+        @ConfigItem(defaultValue = "true")
+        boolean enableUi;
+    }
+
+    @BuildStep
+    void feature(BuildProducer<FeatureBuildItem> featureProducer) {
+        featureProducer.produce(new FeatureBuildItem(FeatureBuildItem.SMALLRYE_GRAPHQL));
+    }
+
+    @BuildStep
+    void additionalBeanDefiningAnnotation(BuildProducer<BeanDefiningAnnotationBuildItem> beanDefiningAnnotationProducer) {
+        // Make ArC discover the beans marked with the @GraphQlApi qualifier
+        beanDefiningAnnotationProducer.produce(new BeanDefiningAnnotationBuildItem(Annotations.GRAPHQL_API));
+    }
+
+    @BuildStep
+    void additionalBean(BuildProducer<AdditionalBeanBuildItem> additionalBeanProducer) {
+        additionalBeanProducer.produce(AdditionalBeanBuildItem.builder()
+                .addBeanClass(GraphQLConfig.class)
+                .addBeanClass(GraphQLProducer.class)
+                .setUnremovable().build());
+    }
+
+    @BuildStep
+    void registerNativeImageResources(BuildProducer<ServiceProviderBuildItem> serviceProvider) throws IOException {
+        // Lookup Service (We use the one from the CDI Module)
+        String lookupService = SPI_PATH + LookupService.class.getName();
+        Set<String> lookupImplementations = ServiceUtil.classNamesNamedIn(Thread.currentThread().getContextClassLoader(),
+                lookupService);
+        serviceProvider.produce(
+                new ServiceProviderBuildItem(LookupService.class.getName(), lookupImplementations.toArray(new String[0])));
+
+        // Metrics (We use the one from the CDI Module)
+        String metricsService = SPI_PATH + MetricsService.class.getName();
+        Set<String> metricsImplementations = ServiceUtil.classNamesNamedIn(Thread.currentThread().getContextClassLoader(),
+                metricsService);
+        serviceProvider.produce(
+                new ServiceProviderBuildItem(MetricsService.class.getName(), metricsImplementations.toArray(new String[0])));
```

Comment:
Is this related to SmallRye Metrics, should we test the presence of it?

Reply:
Metrics integration is not supposed to be covered by this PR yet, I am planning to add that later, so I think this is some leftover that can be removed for now

Reply:
No, I added it to load the SPI (Native mode). @jmartisk will this use another service implementation ? I can remove this, then you add it when you do the metrics ?

Reply:
Well, yeah, this is just one part of the metrics integration - without the rest, the metrics won't work properly anyway. So yeah I'd say we can remove these six lines for now, I'll do metrics separately later and re-add this